locust/test/test_main.py

```diff
@@ -1429,15 +1952,51 @@ def my_task(self):
                 stderr=PIPE,
                 text=True,
             )
-            gevent.sleep(1.9)
-            proc.send_signal(signal.SIGINT)
-            stdout, stderr = proc.communicate()
-            print(stderr, stdout)
-            self.assertIn("Shape test starting", stderr)
-            self.assertIn("Exiting due to CTRL+C interruption", stderr)
-            self.assertIn("Test Stopped", stdout)
-            # ensure stats printer printed at least one report before shutting down and that there was a final report printed as well
-            self.assertRegex(stderr, r".*Aggregated[\S\s]*Shutting down[\S\s]*Aggregated.*")
+
+            def poll_for_message(expected_message, timeout=10):
+                """Polls stderr until the expected message appears or the timeout is reached."""
+                start_time = time.time()
+                buffer = ""
+
+                while time.time() - start_time < timeout:
+                    ready, _, _ = select.select([proc.stderr], [], [], 0.1)
+                    if ready:
+                        line = proc.stderr.readline()
+                        buffer += line
+                        if expected_message in line:
+                            return True
+                    gevent.sleep(0.1)  # Prevent CPU hogging
+                return False
```

Comment:
For this `poll_for_message` function, could we use the top level `poll_until` with a callback that checks stderr for the expected message?